src/Component/CardStyle/CardStyle.tsx

```diff
@@ -82,131 +76,318 @@ interface CardStyleDefaultProps {
+    let styleClone = _cloneDeep(style);
+    const ruleIdx = currentView.path[currentView.path.length - 1].indices[0];
+    const symbolizerIdx = currentView.path[currentView.path.length - 1].indices[1];
+    styleClone.rules[ruleIdx].symbolizers[symbolizerIdx] = symbolizer;
+    if (onStyleChange) {
+      onStyleChange(styleClone);
+    }
+  };
+
+  const onFilterChange = (filter: GsFilter) => {
+    let styleClone = _cloneDeep(style);
+    const ruleIdx = currentView.path[currentView.path.length - 1].indices[0];
+    styleClone.rules[ruleIdx].filter = filter;
+    if (onStyleChange) {
+      onStyleChange(styleClone);
+    }
+  };
+
+  const onBulkEditorChange = (prop: string, val: any) => {
+    const styleClone = _cloneDeep(style);
+    const selectedRuleIds = currentView.path[currentView.path.length - 1].indices;
+    styleClone.rules
+      .filter((rule: GsRule, idx: number) => {
+        return selectedRuleIds.includes(idx);
+      })
+      .forEach((rule: GsRule) => {
+        rule.symbolizers.forEach((sym: any) => {
+          sym[prop] = val;
+          if (prop === 'image' && sym.wellKnownName) {
+            delete sym.wellKnownName;
+            sym.kind = 'Icon';
+          }
+          if (prop === 'wellKnownName' && sym.image) {
+            delete sym.image;
+          }
+        });
+      });
+
+    if (onStyleChange) {
+      onStyleChange(styleClone);
+    }
+  };
+
+  const onIconSelect = (iconSrc: string) => {
+    const styleClone = _cloneDeep(style);
+    const ruleIdx = currentView.path[currentView.path.length - 1].indices[0];
+    const symbolizerIdx = currentView.path[currentView.path.length - 1].indices[1];
+    const symbolizer = styleClone.rules[ruleIdx].symbolizers[symbolizerIdx];
+    if (isIconSymbolizer(symbolizer)) {
+      symbolizer.image = iconSrc;
+    }
+
+    if (onStyleChange) {
+      onStyleChange(styleClone);
+    }
+  };
+
+  /**
+   * Get the image url for the currently selected symbolizer.
+   */
+  const getSelectedIconSrc = (): string => {
+    const symbolizer = style
+      .rules[currentView.path[currentView.path.length - 1].indices[0]]
+      .symbolizers[currentView.path[currentView.path.length - 1].indices[1]];
+
+    if (isIconSymbolizer(symbolizer)) {
+      return symbolizer.image;
+    }
+    return '';
+  };
+
+  const onRuleChangeView = (viewName: string, indices: number[]) => {
+    const newIndices = [currentView.path[currentView.path.length - 1].indices[0], ...indices];
+    changeView(viewName, newIndices);
+  };
+
+  const onIconEditorChangeView = () => {
+    const newIndices = [...currentView.path[currentView.path.length - 1].indices];
+    changeView(ICONLIBRARIESVIEW, newIndices);
+  };
+
+  const styleOverviewPropsOverwrites = {
+    rulesProps: {
+      ruleCardProps: {
+        rendererProps: {
+          rendererType: rendererType
+        }
+      }
+    }
+  };
+  const mergedStyleOverviewProps = _merge(styleOverviewProps, styleOverviewPropsOverwrites);
```

Comment:
No lodas needed herer i guess.
```suggestion
  const mergedStyleOverviewProps = {...styleOverviewProps, ...styleOverviewPropsOverwrites};
```